src/controllers/expenses.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const userService = require('../services/users');
+const expensesService = require('../services/expenses');
+
+const getAllExpenses = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  let expenses = expensesService.getAllExpenses();
+
+  if (userId) {
+    expenses = expenses.filter((elem) => elem.userId === +userId);
+  }
+
+  if (categories) {
+    expenses = expenses.filter((elem) => elem.category === categories);
+  }
+
+  if (from && to) {
+    expenses = expenses.filter(({ spentAt }) => {
+      const expanseDate = new Date(spentAt);
+      const fromDate = new Date(from);
+      const toDate = new Date(to);
+
+      return fromDate <= expanseDate && toDate > expanseDate;
+    });
+  }
+
+  res.send(expenses);
+};
+
+const getOneExpense = (req, res) => {
+  const { expensesId } = req.params;
+  const regex = /^\d+$/;
+
+  if (!regex.test(expensesId)) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const expense = expensesService.findExpensesById(expensesId);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const addExpense = (req, res) => {
+  const data = req.body;
+  const userExpense = userService.findUserById(data.userId);
+
+  if (!Object.entries(data).length || !userExpense) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = expensesService.createExpense(data);
+
+  res.statusCode = 201;
+  res.send(newExpense);
```

Comment:
Use 'res.status(201).send(newExpense);' instead of setting the status code separately.
```suggestion
  res.status(201).send(newExpense);
```

Reply:
fix it in all cases